lib/formtools/form.js

```diff
@@ -225,14 +232,11 @@ var generateFormFromModel = exports.generateFormFromModel = function (m, schemaF
 
                 } else {
 
-                    switch (field.type) {
+                    switch (fieldType) {
 
                         case 'date':
-                        fieldValue = ((r[fieldName] !== undefined && r[fieldName] !== null) ? moment(r[fieldName]).format('YYYY-MM-DD') : null) || defaultValue;
-                        break;
-
                         case 'datetime':
-                        fieldValue = ((r[fieldName] !== undefined && r[fieldName] !== null && (r[fieldName]) instanceof Date) ? r[fieldName].toISOString().slice(0,23) : defaultValue) || defaultValue;
+                            fieldValue = r[fieldName] !== undefined && r[fieldName] !== null ? moment(r[fieldName]).toISOString() : defaultValue;
```

Comment:
Have you tested for this change specifically (i.e. `date` and `datetime`)?

Reply:
Yeah, there is a bit of a problem here though.
We have been updating the date widget, but not the datetime or datetimelocal widget. I think neither of these have worked for a while.

Did you want me to fix those up or remove them completely from Linz? I don't see much point in keeping them since the date widget can handle everything that these other widgets can.